packages/product-json-to-csv/src/cli.js

```diff
@@ -0,0 +1,239 @@
+import fs from 'fs'
+import readline from 'readline'
+import { getCredentials } from '@commercetools/get-credentials'
+import pino from 'pino'
+import PrettyError from 'pretty-error'
+import yargs from 'yargs'
+
+import ProductJsonToCsv from './main'
+import { description } from '../package.json'
+
+process.title = 'product-json-to-csv'
+
+const args = yargs
+  .usage(
+    `
+Usage: $0 [options]
+${description}`
+  )
+  .showHelpOnFail(false)
+  .help()
+  .alias('help', 'h')
+  .version()
+  .alias('version', 'v')
+  .option('projectKey', {
+    alias: 'p',
+    describe: 'API project key',
+    demandOption: true,
+    type: 'string',
+  })
+  .option('apiUrl', {
+    default: 'https://api.commercetools.com',
+    describe: 'The host URL of the HTTP API service',
+    type: 'string',
+  })
+  .option('authUrl', {
+    default: 'https://auth.commercetools.com',
+    describe: 'The host URL of the OAuth API service',
+    type: 'string',
+  })
+  .option('accessToken', {
+    describe: `CTP client access token
+Required scopes: ['view_products']`,
+    type: 'string',
+  })
+  .option('template', {
+    alias: 't',
+    describe:
+      'CSV file containing your header that defines what you want to export.',
+  })
+  .coerce('template', arg => {
+    if (fs.existsSync(arg)) {
+      if (arg.match(/\.csv$/i)) return fs.createReadStream(String(arg))
+
+      throw new Error('Invalid file format. Must be CSV file')
+    }
+    throw new Error('File cannot be reached or does not exist')
+  })
+  .option('input', {
+    alias: 'i',
+    default: 'stdin',
+    describe: 'Path from which to read product chunks.',
+  })
+  .coerce('input', arg => {
+    if (arg === 'stdin') return process.stdin
+
+    if (fs.existsSync(arg)) {
+      if (arg.match(/\.json$/i)) return fs.createReadStream(String(arg))
+
+      throw new Error('Invalid input file format. Must be JSON file')
+    }
+    throw new Error('Input file cannot be reached or does not exist')
+  })
+  .option('output', {
+    alias: 'o',
+    default: 'stdout',
+    describe: 'Path to output',
+    type: 'string',
+  })
+  .coerce('output', arg => {
+    if (arg !== 'stdout') return fs.createWriteStream(String(arg))
+
+    return process.stdout
+  })
+  .option('categoryBy', {
+    choices: ['name', 'key', 'externalId', 'namedPath'],
+    default: 'name',
+    describe:
```

Comment:
Man, it has been a while since I worked on such a tool 😅 What does this option do exactly again? 🙈

Does this depend on whether there is a category header in the template file?

Reply:
@PhilippSpo The `categories` column in the exported product can either reference the `name`, `key`, `externalId`, or full `namedPath` of the category

Reply:
Ah ok got it thanks 👍  Maybe naming this `referenceCategoryBy` or something similar would clear things up 💡